contracts/minter-suite/Minters/MinterPolyptych/MinterPolyptychV0.sol

```diff
@@ -0,0 +1,821 @@
+         * @dev Considered an interaction because calling ownerOf on an NFT
+         * contract. Plan is to only register AB/PBAB NFTs on the minter, but
+         * in case other NFTs are registered, better to check here. Also,
+         * function is non-reentrant, so being extra cautious.
+         */
+        if (msg.sender == _artist) {
+            require(
+                IERC721(_ownedNFTAddress).ownerOf(_ownedNFTTokenId) == _to,
+                "Only owner of NFT"
+            );
+        } else {
+            require(
+                IERC721(_ownedNFTAddress).ownerOf(_ownedNFTTokenId) == vault,
+                "Only owner of NFT"
+            );
+        }
+
+        // split funds
+        _splitFundsETH(_projectId, _pricePerTokenInWei);
+
+        return tokenId;
+    }
+
+    /**
+     * @dev splits ETH funds between sender (if refund), foundation,
+     * artist, and artist's additional payee for a token purchased on
+     * project `_projectId`.
+     * @dev possible DoS during splits is acknowledged, and mitigated by
+     * business practices, including end-to-end testing on mainnet, and
+     * admin-accepted artist payment addresses.
+     */
+    function _splitFundsETH(
+        uint256 _projectId,
+        uint256 _pricePerTokenInWei
+    ) internal {
+        if (msg.value > 0) {
+            bool success_;
+            // send refund to sender
+            uint256 refund = msg.value - _pricePerTokenInWei;
+            if (refund > 0) {
+                (success_, ) = msg.sender.call{value: refund}("");
+                require(success_, "Refund failed");
+            }
+            // split remaining funds between foundation, artist, and artist's
+            // additional payee
+            (
+                uint256 renderProviderRevenue_,
+                address payable renderProviderAddress_,
+                uint256 platformProviderRevenue_,
+                address payable platformProviderAddress_,
+                uint256 artistRevenue_,
+                address payable artistAddress_,
+                uint256 additionalPayeePrimaryRevenue_,
+                address payable additionalPayeePrimaryAddress_
+            ) = genArtCoreContract.getPrimaryRevenueSplits(
+                    _projectId,
+                    _pricePerTokenInWei
+                );
+            // Render provider (e.g. Art Blocks) payment
+            if (renderProviderRevenue_ > 0) {
+                (success_, ) = renderProviderAddress_.call{
+                    value: renderProviderRevenue_
+                }("");
+                require(success_, "Render provider payment failed");
+            }
+            // Platform provider (e.g. engine partner) payment
+            if (platformProviderRevenue_ > 0) {
+                (success_, ) = platformProviderAddress_.call{
+                    value: platformProviderRevenue_
+                }("");
+                require(success_, "Platform provider payment failed");
+            }
```

Comment:
The lines above ensure we handle the platform payment splits on the engine contract.